### PR TITLE
Export metrics to Prometheus

### PR DESCRIPTION
**Pull request by johngmyers**
The format specification is at https://prometheus.io/docs/instrumenting/exposition_formats/

### CODE REVIEW

**Comment by johngmyers** [comment]
Note that GitHub's web UI is showing the commits in the wrong order. You might want to look at this in an IDE.

File: reporting-prometheus/src/test/java/com/proofpoint/reporting/TestReportingPrometheusModule.java [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+                prepareGet().setUri(uriFor("/metrics")).build(),
+                createStringResponseHandler());
+
+        assertEquals(response.getStatusCode(), 200);
+        assertEquals(response.getBody(),
+                "#TYPE ReportCollector_NumMetrics gauge\n" +
+                        "ReportCollector_NumMetrics 2\n" +
+                        "#TYPE TestObject_Metric gauge\n" +
+                        "TestObject_Metric{a=\"b\",baz=\"quux\",c=\"d\\\"\\\\\\n\"} 2\n" +
+                        "TestObject_Metric{foo=\"bar\"} 2\n");
```
Comment (gwittel) [inline reporting-prometheus/src/test/java/com/proofpoint/reporting/TestReportingPrometheusModule.java]: Minor: Do you care about testing the proper handling of labels with invalid chars as well?

Comment (johngmyers) [reply reporting-prometheus/src/test/java/com/proofpoint/reporting/TestReportingPrometheusModule.java]: Ah, I could do it with `@Prometheus`, which didn't exist at the time.

**Comment by johngmyers** [comment]
Rebased and merged

**Pull request closed without merging**